Added code in app.js that updates urlParams appropriately on back and forward browser button presses

gitairman opened the pull request:
Fixes Issue #39 

https://github.com/sahajsk21/Anvesha/assets/97379449/5d71f2ae-bb58-43f0-b80f-32283a2f13c1


https://github.com/sahajsk21/Anvesha/assets/97379449/618e0dfa-cde2-4e26-a858-255940bbe8fb

gitairman committed:
Added packages to be able to run Vite/Vue

gitairman committed:
update urlParams based current value and href

gitairman committed:
Adding dist

gitairman committed:
for adding to github pages

gitairman committed:
modified workaround for browser button presses

gitairman committed:
to enable app in github pages

gitairman committed:
updated config.js

gitairman committed:
added comments on line 89

gitairman committed:
Removed troubleshooting code

gitairman committed:
Removed unncessary code comment

gitairman committed:
Merge branch 'sahajsk21:master' into master

gitairman committed:
updated excluded files and folders

gitairman committed:
update excluded files

gitairman committed:
Merge branch 'master' of https://github.com/gitairman/Anvesha

gitairman committed:
Delete .github directory

gitairman committed:
Delete .DS_Store

gitairman committed:
Delete node_modules directory

gitairman committed:
Delete dist directory

gitairman committed:
Delete .gitlab-ci.yml

gitairman committed:
Delete vite.config.js

gitairman committed:
Delete package.json

gitairman committed:
Delete package-lock.json

yaronkoren commented:
@gitairman - this looks like a great fix!

I do have some comments:

- The "Fixes issue #39" thing should go in the commit description, not the commit title - the commit title should actually describe the fix.
- It looks like only about the first 10 lines are actually code changes - the other changed lines are just space reformatting, and that sort of thing. Could you get rid of those other changes? They may be more correct formatting, but they make it hard to see what changed in this patch.
- Could you put in some comment to explain why, if the in-memory URL parameters don't change, the URL parameters variable needs to be reset? I think I understand this, but for people reading through the code, it might not be clear.
- Actually, on that note: maybe it's worth just resetting the URL parameters variable every time, no matter what, without checking anything? Or is this not a good idea, for one reason or another? You could argue that it's an unnecessary performance hit - but on the other hand, going through the "current" and "previous" variables to check if they're the same every time is a performance hit, so maybe there's no big difference. Although maybe that wouldn't work.

gitairman commented:
> @gitairman - this looks like a great fix!
> 
> I do have some comments:
> 
> * The "Fixes issue 39" thing should go in the commit description, not the commit title - the commit title should actually describe the fix. **_--updated_**
> * It looks like only about the first 10 lines are actually code changes - the other changed lines are just space reformatting, and that sort of thing. Could you get rid of those other changes? They may be more correct formatting, but they make it hard to see what changed in this patch. _**--I think this happened because VScode may have autoformatted the file as I was editing it_**
> * Could you put in some comment to explain why, if the in-memory URL parameters don't change, the URL parameters variable needs to be reset? I think I understand this, but for people reading through the code, it might not be clear. **_--Shall I put this comment above the code in app.js or in the description of the pull request?_**
> * Actually, on that note: maybe it's worth just resetting the URL parameters variable every time, no matter what, without checking anything? **_--I'm still not clear about what functions get executed, and in which order, on the back and forward browser button presses.  I added the new code in the window.onpopstate function as that seemed to be the only function within app.js that executed on a back or forward button press.  However, updating the urlParams object could probably be done in another place in app.js or index.js_**

Or is this not a good idea, for one reason or another? You could argue that it's an unnecessary performance hit - but on the other hand, going through the "current" and "previous" variables to check if they're the same every time is a performance hit, so maybe there's no big difference. Although maybe that wouldn't work.**_--the urlParams object seems to only be updated on anchor clicks and not on browser back/fwd presses.  I did go through an iteration of updating urlParams on every back/fwd press.  However, I ran into some problems doing that because there has to be some other variable to update urlParams with, and the only other one I could think of (without creating any) was the window.location.href value.  And in the function I was working, the href value doesn't update until after, so when I updated the urlParams every time with href there was some weird behaviour.  

Checking the urlParams against a previous version of itself basically lets us know that there was a back OR fwd button press because these seem to be the only situations when it is not updated in an already existing method.  So when the current and previous urlParams are identical, this means back or fwd was pressed, in which case this is the only time it is updated with the values extracted from the historically logged href string.

I'm certain there is a more elegant way to achieve this fix, but my current knowledge of the codebase and javaScript in general is still quite limited._**
